moto/iam/models.py

```diff
@@ -937,5 +948,33 @@ def get_account_authorization_details(self, filter):
             'managed_policies': returned_policies
         }
 
+    def create_saml_provider(self, name, saml_metadata_document):
+        saml_provider = SAMLProvider(name, saml_metadata_document)
+        self.saml_providers[name] = saml_provider
+        return saml_provider
+
+    def update_saml_provider(self, saml_provider_arn, saml_metadata_document):
+        saml_provider = self.get_saml_provider(saml_provider_arn)
+        saml_provider.saml_metadata_document = saml_metadata_document
+        return saml_provider
+
+    def delete_saml_provider(self, saml_provider_arn):
+        try:
+            for saml_provider in list(self.list_saml_providers()):
```

Comment:
This was done because it made more sense (chicken and egg) in my opinion to store the objects by their `Name` when they get created. By virtue of that, we can't query on the ARN for the object intrinsically.

I had to cast this as a `list()` so that way it makes a copy, otherwise we get a race condition:

```
RuntimeError: dictionary changed size during iteration
```